File: c++/C++YT/rotate_array.cpp
// Given an array, rotate the array to the right by k steps, where k is non-negative.
#include<iostream>
using namespace std;
void rotate(int nums[], int size, int k)
{
    int temp[size]; 
    for(int i=0; i<size; i++)
    {
        temp[(i+k)%size]=arr[i];
    }
    arr = temp;
}
void printArray(int ans[],int size)
{
    for(int i=0; i<size; i++)
    {
        cout<<ans[i]<<" ";
    }
}
int main()
{
    vector<int> nums{ 10, 20, 30 };
    cout<<"enter steps after which you want to rotate:";
    cin>>k;
    rotate(nums, 6, k);
    printArray(arr, 6);
}

File: java/array/Rotate_array.java
package array;
public class Rotate_array 
{
    static void rotate(int[] arr, int size, int k)
    {
        int[] temp= new int[size];
        for(int i=0; i<size; i++)
        {
            temp[(i+k)%size] = arr[i];
        }
        for(int i=0; i<size; i++)
        System.out.print(temp[i]+" ");
    }
    public static void main(String[] args)
    {
        int[] arr=  { 10, 20, 30, 40, 50, 60, 70 };
        int size=arr.length;
        int key=3;
        rotate(arr, size, key);
    }    
}




// public class Rotate_array 
// {
//     static void rotate(int[] arr, int size, int k)
//     {
//         int[] temp = new int[size];
//         for(int i=0; i<size; i++){
//             temp[(i+k)%size] = arr[i];
//         }
//         for(int i=0; i<size; i++){
//             System.out.print(temp[i]+" ");
//         }
//     }
//     public static void main(String[] args)
//     {
//         int[] arr=  { 10, 20, 30, 40, 50, 60, 70 };
//         int size=arr.length;
//         int key=3;
//         rotate(arr, size, key);
//     }    
// }
